### ISSUE TITLE
Spectrum1D is not able to take 2D WCS for 2D dataset

### ISSUE THREAD

**Issue by astrofrog** [astropy/specutils#857]
I am trying to provide a 2D WCS to the Spectrum1D class but this does not work:

```python
import numpy as np
from astropy.wcs import WCS
from astropy import units as u
from specutils import Spectrum1D

wcs = WCS(naxis=2)
wcs.wcs.ctype = ['SPATIAL', 'FREQ']
wcs.wcs.set()

flux = np.ones((10, 15)) * u.Unit('Jy')

spec2d = Spectrum1D(flux, wcs=wcs)
```

This is despite the axes being completely separable in the WCS. Based on discussions with @eteq I would have thought this would be supported?

**Comment by eteq** [comment]
Yeah, I'd say that should work too. Note that it's not just sufficient that it be separable, the spectral axes should be the *same values* for all the other parts of the cube (that's what distinguishes `Spectrum1D` from `SpectrumCollection`), but if you're providing a wcs `specutils` should accept it so that you can provide the spatial information...

(Whether the initializer should *check* if the `Spectrum1D` assumption is true versus taking a "let the user beware" attitude is debatable, but it certainly should be possible to pass in a multi-D WCS that satisfies the assumptions of Spectrum1D )

**Comment by eteq** [comment]
(oh, it might also need to have spectral-compatible *units* on the freq axis, not just the ctype marker.  But that's not relevant to the larger problem that a 2d wcs isn't accepted)
